vscode-lean4/test/suite/toolchains/toolchain.test.ts

```diff
@@ -3,12 +3,56 @@ import { suite } from 'mocha';
 import * as path from 'path';
 import * as vscode from 'vscode';
 import * as fs from 'fs';
-import { initLean4Untitled, initLean4, waitForInfoviewHtml, closeAllEditors,
-	extractPhrase, restartLeanServer, assertStringInInfoview, resetToolchain } from '../utils/helpers';
+import { initLean4Untitled, initLean4, waitForInfoviewHtml, closeAllEditors, assertActiveClient,
+	extractPhrase, restartLeanServer, assertStringInInfoview, resetToolchain, insertText, deleteAllText } from '../utils/helpers';
 
 // Expects to be launched with folder: ${workspaceFolder}/vscode-lean4/test/suite/simple
 suite('Toolchain Test Suite', () => {
 
+	test('Worker crashed and client running', async () => {
+		console.log('=================== Test worker crashed and client running ===================');
+		void vscode.window.showInformationMessage('Running tests: ' + __dirname);
+
+		// add normal values to initialize lean4 file
+		const hello = 'Hello World'
+		const lean = await initLean4Untitled(`#eval "${hello}"`);
+		const info = lean.exports.infoProvider;
+		assert(info, 'No InfoProvider export');
+
+		console.log('make sure language server is up and running.');
+		await assertStringInInfoview(info, hello);
+
+		const clients = lean.exports.clientProvider;
+		assert(clients, 'No LeanClientProvider export');
+
+		console.log('Insert eval that causes crash.')
+		await insertText('\n\n#eval (unsafeCast 0 : String)')
+
+		const expectedMessage = 'The Lean Server has stopped processing this file'
+		await assertStringInInfoview(info, expectedMessage);
+
+		console.log('restart the server (without modifying the file, so it should crash again)')
+		let client = assertActiveClient(clients);
+		await restartLeanServer(client);
+
+		console.log('Checking that it crashed again.')
+		await assertStringInInfoview(info, expectedMessage);
+
+		console.log('deleting the problematic string closing active editors and restarting the server')
+		await deleteAllText();
+		await insertText(`#eval "${hello}"`);
+		console.log('Now invoke the restart server command')
+		client = assertActiveClient(clients);
+		await restartLeanServer(client);
```

Comment:
Let's make this one a "restartFile" command so we have some test coverage that this works also.